ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -358,39 +358,66 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
-                                                                    preferencesKey='application-details'>
-                                                                    {data => (
-                                                                        <ApplicationResourceList
-                                                                            onNodeClick={fullName => this.selectNode(fullName)}
-                                                                            resources={data}
-                                                                            nodeMenu={node =>
-                                                                                AppUtils.renderResourceMenu(
-                                                                                    {...node, root: node},
-                                                                                    application,
-                                                                                    tree,
-                                                                                    this.appContext,
-                                                                                    this.appChanged,
-                                                                                    () => this.getApplicationActionMenu(application, false)
-                                                                                )
-                                                                            }
-                                                                        />
+                                                    <DataLoader
+                                                        input={{filteredRes}}
+                                                        load={async () => {
+                                                            const liveStatePromises = filteredRes.map(async resource => {
+                                                                const resourceRow: any = {...resource, group: resource.group || ''};
+                                                                const liveState =
+                                                                    typeof resource.group !== 'undefined' &&
+                                                                    (await services.applications.getResource(application.metadata.name, resource).catch(() => null));
+                                                                if (liveState?.metadata?.annotations && liveState?.metadata?.annotations[models.AnnotationHookKey]) {
+                                                                    resourceRow.syncOrder = liveState?.metadata.annotations[models.AnnotationHookKey];
+                                                                    if (liveState?.metadata?.annotations && liveState?.metadata?.annotations[models.AnnotationSyncWaveKey]) {
```

Comment:
i think this can be changed to liveState?.metadata?.annotations?.[models.AnnotationSyncWaveKey]

Reply:
Thanks for the advice! This works great 👍